php/routes.php

```diff
@@ -26,4 +26,42 @@
             ]
         );
     });
+
+    CBPRouter::code('/code', function ($payload) {
+        $parsedUrl = wp_parse_url($payload['url']);
+
+        $isValidHost = $parsedUrl['host'] === 'gist.githubusercontent.com' ||
+            $parsedUrl['host'] === 'raw.githubusercontent.com' ||
+            $parsedUrl['host'] === 'github.com';
+
+        if (!$isValidHost) {
+            return new WP_REST_Response(
+                [
+                    'code' => 'Error: Invalid Host'
+                ]
+            );
+        }
+
+        if ($parsedUrl['host'] === 'github.com') {
+            $parsedUrl['host'] = 'raw.githubusercontent.com';
+            $parsedUrl['path'] = str_replace('/blob', '', $parsedUrl['path']);
+            $payload['url'] = $parsedUrl['scheme'] . '://' . $parsedUrl['host'] . $parsedUrl['path'];
+        }
+
+        $response = wp_remote_get($payload['url']);
+
+        if ($response['headers']['content-type'] !== 'text/plain; charset=utf-8') {
+            return new WP_REST_Response(
+                [
+                    'code' => 'Error: Invalid Content-Type'
+                ]
+            );
+        }
+
+        return new WP_REST_Response(
+            [
+                'code' => $response['body']
+            ]
+        );
+    });
```

Comment:
I think if we're all good we can just send back the code without the key, also, better to use wp's function here

```php
return new \WP_REST_Response(wp_remote_retrieve_body($response));
```